### PR TITLE
Enhance `can.logger` to consider the `append` option

### LINKED ISSUE #1326: Enhance the `can.logger` to check to see if the log file exists and append if there rather than overwrite
### Is your feature request related to a problem? Please describe.

Consider `can.logger` being used on an embedded Linux system. A `python can.logger ...` process is started as a daemon (`/etc/init.d/`) in boot and killed in shutdown. When the system is restarted, the existing log file on the device and its contents is currently overwritten. This results in a loss of data. 

### Describe the solution you'd like

Upon executing `python can.logger ...` the program should check to see if the specified log file exists. If it does exist, then the file should be appended to. 

### Describe alternatives you've considered

An optional argument could be added (perhaps `-a` or `-w`) to give the user the option as to whether an existing file, if found, should be overwritten or appended to. 

### Additional context

P.S. Feel free to assign this issue to me. I should be around to it by Sunday or Monday.

### PR DESCRIPTION
**Pull request by j-c-cook**
The goal here appears to be to give access to the boolean `append` option of `Printer`. 

https://github.com/hardbyte/python-can/blob/0d7f65f74b8b015fb650e9910a920323023f310d/can/io/printer.py#L16-L40

closes #1326

### CODE REVIEW

File: can/logger.py [hunk 1]
```diff
@@ -201,12 +211,36 @@ def main() -> None:
...
+                    action = args[j+1]
+                else:
+                    break
+                # Consider the action, and update the results to accurately embody request
+                if action == '0' or action == 'False':
+                    results.append_mode = False
+                elif action == '1' or action == 'True':
+                    results.append_mode = True
+                else:
+                    pass
```
Comment (j-c-cook) [inline can/logger.py:235]: I added a manual `argparse` to search for `boolean_args` and correct the `results` if necessary. 

NOTE: Only required until Python>=3.9
NOTE: Please do not use this as a reason to advocate Python>=3.9.

**Comment by j-c-cook** [comment]
@hardbyte @zariiii9003 This accomplishes what was descriped in #1327 and is ready for review.

**Comment by zariiii9003** [comment]
Could you fix the linters?

**Comment by j-c-cook** [comment]
@zariiii9003 The `*lack` format is failing here (https://github.com/hardbyte/python-can/runs/6761006357?check_suite_focus=true), but when I run it locally the checks pass. 

```
$ black --check --verbose logger.py
Identified `/home/user/Documents/python-can` as project root containing a .git directory.
Sources to be formatted: "can/logger.py"
logger.py wasn't modified on disk since last run.

All done! ✨ 🍰 ✨
1 file would be left unchanged.
```

**Comment by zariiii9003** [comment]
I guess you are using a different black version. Run `pip install -r requirements-lint.txt`

**Comment by j-c-cook** [comment]
@zariiii9003 Do you want for me to add in tests to try to resolve the https://github.com/hardbyte/python-can/pull/1327/checks?check_run_id=6775694757 failure?

File: can/logger.py [hunk 2]
```diff
@@ -201,12 +219,42 @@ def main() -> None:
     print(f"Connected to {bus.__class__.__name__}: {bus.channel_info}")
     print(f"Can Logger (Started on {datetime.now()})")
 
+    # argparser.py in Python 3.7.11 does not properly consider boolean
```
Comment (zariiii9003) [inline can/logger.py:222]: You don't need any of this. Look at how the `--fd` argument is handled. It is also a boolean.

Comment (j-c-cook) [reply can/logger.py]: I agree. I have modified my changes. This simplifies things. Thank you.

File: can/logger.py [hunk 3]
```diff
@@ -154,12 +157,20 @@ def main() -> None:
         default=None,
     )
 
+    parser.add_argument(
+        "--a",
```
Comment (zariiii9003) [inline can/logger.py:161]:
```suggestion
        "-a",
```

File: can/logger.py [hunk 4]
```diff
@@ -154,12 +157,20 @@ def main() -> None:
...
+        "--append",
```
Comment (zariiii9003) [inline can/logger.py:162]:
```suggestion
        "--append",
        dest="append",
```

File: can/logger.py [hunk 5]
```diff
@@ -201,12 +212,13 @@ def main() -> None:
     print(f"Connected to {bus.__class__.__name__}: {bus.channel_info}")
     print(f"Can Logger (Started on {datetime.now()})")
 
+    options = {"append": results.a}
```
Comment (zariiii9003) [inline can/logger.py:215]:
```suggestion
    options = {"append": results.append}
```

**Review by zariiii9003 — CHANGES_REQUESTED** [review]

**Comment by j-c-cook** [comment]
@zariiii9003 I committed your recommendations. Next time you do that I'll make sure to accept them from this UI rather than manually change them.

**Review by zariiii9003 — APPROVED** [review]

**Pull request merged**